tests/init/neg/early-promote5.scala

```diff
@@ -1,5 +1,5 @@
 abstract class A {
-  bar(this)
+  bar(this) // error
```

Comment:
Why we don't have an error before?

Reply:
Previously, the `fields` in `tryPromote` doesn't include c in `class M(val o: Outer, c: Container) extends A with o.B`, so promoting `val m = new M(o, this)` doesn't trigger the error